### ISSUE TITLE
Filter out retracted versions by default

### ISSUE THREAD

**Issue by wjam** [icholy/gomajor#32]
Some Go projects, such as [the Kubernetes client-go library](https://github.com/kubernetes/client-go/blob/v1.5.2/go.mod), have [retracted](https://go.dev/doc/modules/gomod-ref#retract) some versions of the library. In the case of `client-go`, this means that _no_ `v1.x` version should be used.

With this `go.mod` file:

```
module demo

go 1.22.0

toolchain go1.22.2

require k8s.io/client-go v0.30.2
```

The current behaviour is:

```
$ gomajor list -major
k8s.io/client-go: v0.30.2 [latest v1.5.2]
```

The expected behaviour is:
```
$ gomajor list -major
```

**Comment by icholy** [comment]
Thanks for the report, I'll fix this when I have time.

**Comment by icholy** [comment]
Fixed in v0.12.0
